pkg/commands/config.go

```diff
@@ -61,11 +61,13 @@ func getBaseImage(bo *options.BuildOptions) build.GetBase {
 		//
 		// Platforms can be comma-separated if we only want a subset of the base
 		// image.
-		multiplatform := bo.Platform == "all" || strings.Contains(bo.Platform, ",")
-		if bo.Platform != "" && !multiplatform {
+		allPlatforms := len(bo.Platforms) == 1 && bo.Platforms[0] == "all"
```

Comment:
Do we already have code that fails if you pass `--platform=all,linux/arm64`? Or at least handles it by reducing it to `--platform=all`?

Reply:
I'm not aware of such scenario when I was writing this PR. I thought about adding it to `parseSpec`, but it looked quite complicated, so I decided to preserve the original behavior. Happy to include that into this PR if you'd like to see it!

Reply:
We don't have to include it in this PR, but I'd like to at least handle it better. Maybe another PR.

I've added it to the growing laundry list in https://github.com/google/ko/issues/542.